datafusion/common/src/hash_utils.rs

```diff
@@ -322,7 +322,11 @@ fn hash_fixed_list_array(
 ) -> Result<()> {
     let values = Arc::clone(array.values());
     let value_len = array.value_length();
-    let offset_size = value_len as usize / array.len();
+    let offset_size = if array.len() == 0 {
+        0
+    } else {
+        value_len as usize / array.len()
+    };
```

Comment:
Would this be equiv to `array.value_length()`?

Reply:
I think value_length is different

Like a `FixedSlizeLIst(3)` would have elements that are each 3 bytes, but the array can have any number of elements (rows)

So in that case `value_length()` is 3

Reply:
in this example, value_length() is 3
array.len() is something, say N
so i though that value_len is 3 * N

ie. i kind of suspect this division is calculating what is know from the array type (the 3 in the example)

Reply:
🤔 https://docs.rs/arrow/latest/arrow/array/struct.FixedSizeBinaryArray.html#method.value_length 

I don't understand what this offset_size is really used for to be honest -- it doesn't make a lot of sense below either

It seems the intention is more like `offset_size = array.value_length()` 🤔

Reply:
> It seems the intention is more like `offset_size = array.value_length()` 🤔

that's my take too